src/bin/cargo/commands/check.rs

```diff
@@ -44,7 +44,7 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     let ws = args.workspace(config)?;
     // This is a legacy behavior that causes `cargo check` to pass `--test`.
     let test = matches!(
-        args.get_one::<String>("profile").map(String::as_str),
+        ignore_unknown(args.try_get_one::<String>("profile")).map(String::as_str),
```

Comment:
I wonder why do we need this. `profile` option seems to always be there; or maybe we can use `args._value_of("profile")`.

Reply:
I think this is left over from before I went with #11119